src/smart-components/role/add-role-new/add-permissions.js

```diff
@@ -32,23 +32,42 @@ export const accessWrapper = (rawData, filters = { applications: [], resources:
     );
 
     return {
-        data,
+        data: data.filter(permission => filterSplats(permission)),
         filteredData,
         applications: [ ...new Set(data.filter(permission => filterResource(permission) && filterOperation(permission)).map(({ application }) => application)) ],
         resources: [ ...new Set(data.filter(permission => filterApplication(permission) && filterOperation(permission)).map(({ resource }) => resource)) ],
         operations: [ ...new Set(data.filter(permission => filterApplication(permission) && filterResource(permission)).map(({ operation }) => operation)) ]
     };
 };
 
+export const resolveSplats = (selectedPermissions, permissions) => {
+    return (permissions.length > 0
+        ? selectedPermissions.map(({ uuid: permission }) => {
+            if (permission.includes('*')) {
+                const [ application, resource, operation ] = permission.split(':');
+                return permissions.filter(p => (p.application === application
+                    && (resource === '*' || resource === p.resource)
+                    && (operation === '*' || operation === p.operation))
+                ).map(({ application, resource, operation }) => ({ uuid: `${application}:${resource}:${operation}` }));
+            }
+
+            return { uuid: permission };
+
+        }).flat()
+        : selectedPermissions);
+};
+
 const AddPermissionsTable = ({ selectedPermissions, setSelectedPermissions, ...props }) => {
     const dispatch = useDispatch();
     const fetchData = (apiProps) => dispatch(listPermissions(apiProps));
     const { access, isLoading, pagination, baseRole } = useSelector(selector, shallowEqual);
     const { input } = useFieldApi(props);
     const formOptions = useFormApi();
-    const [ permissions, setPermissions ] = useState({ filteredData: [], applications: [], resources: [], operations: []});
+    const [ permissions, setPermissions ] = useState({ data: [], filteredData: [], applications: [], resources: [], operations: []});
```

Comment:
Might be better to use `useReducer` in future, instead of `useState` with complex state, but that's just a nitpicking, might be good to mark it somehow as code cleenup and left as is.